frontend/src/api/k8s/__tests__/notebooks.spec.ts

```diff
@@ -0,0 +1,936 @@
+  return {
+    ...originalModule,
+    createElyraServiceAccountRoleBinding: jest.fn(),
+  };
+});
+
+const k8sCreateResourceMock = jest.mocked(k8sCreateResource<NotebookKind>);
+const k8sGetResourceMock = jest.mocked(k8sGetResource<NotebookKind>);
+const k8sPatchResourceMock = jest.mocked(k8sPatchResource<NotebookKind>);
+const k8sListResourceMock = jest.mocked(k8sListResource<NotebookKind>);
+const k8sUpdateResourceMock = jest.mocked(k8sUpdateResource<NotebookKind>);
+const k8sDeleteResourceMock = jest.mocked(k8sDeleteResource<NotebookKind, K8sStatus>);
+const createElyraServiceAccountRoleBindingMock = jest.mocked(createElyraServiceAccountRoleBinding);
+
+global.structuredClone = (val: unknown) => JSON.parse(JSON.stringify(val));
+
+const uid = 'test-uid_notebook';
+const username = 'test-user';
+
+describe('assembleNotebook', () => {
+  it('should return NotebookKind object with pipelines', () => {
+    const canEnablePipelines = true;
+
+    const result = assembleNotebook(mockStartNotebookData({}), username, canEnablePipelines);
+    expect(result.apiVersion).toBe('kubeflow.org/v1');
+    expect(result.kind).toBe('Notebook');
+    expect(result.spec.template.spec.volumes).toStrictEqual([
+      { name: 'test-volume', persistentVolumeClaim: { claimName: 'test-volume' } },
+      { name: ELYRA_VOLUME_NAME, secret: { secretName: 'ds-pipeline-config' } },
+      { emptyDir: { medium: 'Memory' }, name: 'shm' },
+    ]);
+    expect(result.spec.template.spec.containers[0].volumeMounts).toStrictEqual([
+      { mountPath: '/opt/app-root/src/data', name: 'test-volume' },
+      { mountPath: '/opt/app-root/runtimes', name: 'elyra-dsp-details' },
+      { mountPath: '/dev/shm', name: 'shm' },
+    ]);
+  });
+  it('should return NotebookKind object with pipelines and volume name is shm', () => {
+    const canEnablePipelines = true;
+
+    const result = assembleNotebook(
+      mockStartNotebookData({ volumeName: 'shm' }),
+      username,
+      canEnablePipelines,
+    );
+
+    expect(result.apiVersion).toBe('kubeflow.org/v1');
+    expect(result.kind).toBe('Notebook');
+
+    expect(result.spec.template.spec.volumes).toStrictEqual([
+      { name: 'shm', persistentVolumeClaim: { claimName: 'shm' } },
+      { name: 'elyra-dsp-details', secret: { secretName: 'ds-pipeline-config' } },
+    ]);
+    expect(result.spec.template.spec.containers[0].volumeMounts).toStrictEqual([
+      { mountPath: '/opt/app-root/src/data', name: 'shm' },
+      { mountPath: '/opt/app-root/runtimes', name: 'elyra-dsp-details' },
+    ]);
+  });
+  it('should return NotebookKind object with pipelines and volume name is elyra-dsp-details', () => {
+    const canEnablePipelines = true;
+
+    const result = assembleNotebook(
+      mockStartNotebookData({ volumeName: ELYRA_VOLUME_NAME }),
+      username,
+      canEnablePipelines,
+    );
+
+    expect(result.apiVersion).toBe('kubeflow.org/v1');
+    expect(result.kind).toBe('Notebook');
+  });
```

Comment:
This test doesn't add any value over the previous set of tests and doesn't match the test description.